### PR TITLE
Gateway REST API Documentation

### PR DESCRIPTION
**Pull request by Sarangan0219**
## Purpose

Adding Docs for the newly introduced Artifact Synchronizer feature 
Related to https://github.com/wso2/product-apim/issues/8246

### CODE REVIEW

File: en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml [hunk 1]
```diff
@@ -0,0 +1,636 @@
...
+  /undeploy-api:
+
+    #-----------------------------------------------------
+    # Undeploy the API in the Gateway
+    #-----------------------------------------------------
+    post:
+      security:
+        - OAuth2Security:
+            - apim:api_publish
+      x-wso2-curl: "curl -k -H \"Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8\" -X POST \"https://127.0.0.1:9443/api/am/gateway/v1/deploy-api?apiId=890a4f4d-09eb-4877-a323-57f6ce2ed79b&apiName=PizzaAPI&label=Food\""
```
Comment (1akshitha) [inline en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml:140]: Don't we have to implement this with the same query params as GET api-artifact resource?  with tenant domain, API name and version.

File: en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml [hunk 2]
```diff
@@ -0,0 +1,636 @@
...
+      tags:
+        - Deploy API
+      responses:
+        200:
+          description: |
+            OK.
+            API successfully deployed in the Gateway.
+          schema:
+            $ref: '#/definitions/DeployResponse'
+        202:
```
Comment (1akshitha) [inline en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml:106]: Have we handled these error codes from impl?

File: en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml [hunk 3]
```diff
@@ -0,0 +1,636 @@
...
+        - OAuth2Security:
+            - apim:api_publish
+      x-wso2-curl: "curl -k -H \"Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8\" -X GET \"https://127.0.0.1:9443/api/am/gateway/v1/api-artifact?apiName=PizzaAPI&version=1.0&tenantDomain=registry.com\""
+      x-wso2-request: |
+        GET https://127.0.0.1:9443/api/am/gateway/v1/api-artifact?apiName=PizzaAPI&version=1.0&tenantDomain=registry.com
+        Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8
+      x-wso2-response: "HTTP/1.1 200 OK"
+      summary: Get API artifact from the storage
+      description: |
+        This operation is used to get the API artifact from the storage
```
Comment (1akshitha) [inline en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml:213]: better to have a meaningful description with all the details returning

File: en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml [hunk 4]
```diff
@@ -0,0 +1,636 @@
...
+    #-----------------------------------------------------
+    # Get local entry for the API
+    #-----------------------------------------------------
+    get:
+      security:
+        - OAuth2Security:
+            - apim:api_publish
+      x-wso2-curl: "curl -k -H \"Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8\" -X GET \"https://127.0.0.1:9443/api/am/gateway/v1/local-entry?apiId=890a4f4d-09eb-4877-a323-57f6ce2ed79b&apiName=PizzaAPI&label=Food\""
+      x-wso2-request: |
+        GET https://127.0.0.1:9443/api/am/gateway/v1/local-entry?apiId=890a4f4d-09eb-4877-a323-57f6ce2ed79b&apiName=PizzaAPI&label=Food
```
Comment (1akshitha) [inline en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml:278]: Don't we have to implement this with the same query params as GET api-artifact resource?  with tenant domain, API name and version.

File: en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml [hunk 5]
```diff
@@ -0,0 +1,636 @@
...
+    #-----------------------------------------------------
+    # Get sequence for the API
+    #-----------------------------------------------------
+    get:
+      security:
+        - OAuth2Security:
+            - apim:api_publish
+      x-wso2-curl: "curl -k -H \"Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8\" -X GET \"https://127.0.0.1:9443/api/am/gateway/v1/sequence?apiId=890a4f4d-09eb-4877-a323-57f6ce2ed79b&apiName=PizzaAPI&label=Food\""
+      x-wso2-request: |
+        GET https://127.0.0.1:9443/api/am/gateway/v1/sequence?apiId=890a4f4d-09eb-4877-a323-57f6ce2ed79b&apiName=PizzaAPI&label=Food
```
Comment (1akshitha) [inline en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml:343]: Don't we have to implement this with the same query params as GET api-artifact resource?  with tenant domain, API name and version.

File: en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml [hunk 6]
```diff
@@ -0,0 +1,636 @@
...
+          description: |
+            Precondition Failed.
+            The request has not been performed because one of the preconditions is not met.
+          schema:
+            $ref: '#/definitions/Error'
+
+  /end-points:
+
+    #-----------------------------------------------------
+    # Get API Artifact
```
Comment (1akshitha) [inline en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml:401]: This should be the endpoint

File: en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml [hunk 7]
```diff
@@ -0,0 +1,636 @@
...
+            End-points successfully retrieved from the storage for the API.
+          schema:
+            $ref: '#/definitions/DeployResponse'
+        202:
+          description: |
+            Accepted.
+            The request has been accepted.
+          schema:
+            $ref: '#/definitions/DeployResponse'
+        400:
```
Comment (1akshitha) [inline en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml:445]: Have we handled all these error codes?

File: en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml (hunk 6 above)
Comment (Sarangan0219) [reply en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml:401]: https://github.com/wso2/docs-apim/pull/1527/commits/7fb0b6cb28a8074e263fe30b1f22a77a2a662d15

File: en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml (hunk 3 above)
Comment (Sarangan0219) [reply en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml]: https://github.com/wso2/docs-apim/pull/1527/commits/7fb0b6cb28a8074e263fe30b1f22a77a2a662d15

File: en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml (hunk 5 above)
Comment (Sarangan0219) [reply en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml]: Fixed https://github.com/wso2/docs-apim/pull/1527/commits/96a86f56bf4414db5ee984eb37e500c62d467e54

File: en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml (hunk 4 above)
Comment (Sarangan0219) [reply en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml]: Fixed https://github.com/wso2/docs-apim/pull/1527/commits/96a86f56bf4414db5ee984eb37e500c62d467e54

File: en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml (hunk 1 above)
Comment (Sarangan0219) [reply en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml:140]: Fixed https://github.com/wso2/docs-apim/pull/1527/commits/96a86f56bf4414db5ee984eb37e500c62d467e54

Comment (Sarangan0219) [reply en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml]: Fixed https://github.com/wso2/docs-apim/pull/1527/commits/2f6d4685e563ba2752305b94dc8bbc50a4335693

File: en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml [hunk 8]
```diff
@@ -0,0 +1,636 @@
...
+  /deploy-api:
+
+    #-----------------------------------------------------
+    # Deploy the API in the Gateway
+    #-----------------------------------------------------
+    post:
+      security:
+        - OAuth2Security:
+            - apim:api_publish
+      x-wso2-curl: "curl -k -H \"Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8\" -X POST \"https://127.0.0.1:9443/api/am/gateway/v1/deploy-api?apiId=890a4f4d-09eb-4877-a323-57f6ce2ed79b&apiName=PizzaAPI&label=Food\""
```
[the remark this answers is not in the document]
Comment (Sarangan0219) [reply en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml]: Fixed https://github.com/wso2/docs-apim/pull/1527/commits/2f6d4685e563ba2752305b94dc8bbc50a4335693

File: en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml (hunk 2 above)
Comment (Sarangan0219) [reply en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml:110]: Handled in rest.api.util

File: en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml (hunk 7 above)
Comment (Sarangan0219) [reply en/docs/develop/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml:465]: Handled in rest.api.util

File: en/docs/develop/product-apis/gateway-apis/gateway-v1/guide.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
+# Getting Started
+
+WSO2 API Manager is a complete solution for publishing APIs, creating and managing a developer community, and for scalably routing API traffic. It leverages proven, production-ready, integration, security and governance components from WSO2 Enterprise Service Bus, WSO2 Identity Server, and WSO2 Governance Registry. Moreover, it is powered by WSO2 Business Activity Monitor, thereby making WSO2 API Manager ready for any large-scale deployments right away.
```
Comment (shaniR) [inline en/docs/develop/product-apis/gateway-apis/gateway-v1/guide.md:3]: We don't use WSO2 ESB, but we use WSO2 EI. We don't use the BAM too. This paragraph needs to be revisted and re written.

Comment (shaniR) [reply en/docs/develop/product-apis/gateway-apis/gateway-v1/guide.md:3]: shall we rephrase it as "WSO2 API Manager is a complete API Management solution  that addresses all aspects of API management including API development, productization, consumption, security, rate limitation, monetization and analytics. "

File: en/docs/develop/product-apis/gateway-apis/gateway-v1/guide.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+
+The REST API is implemented based on REST best practices and specifications as a CXF REST web application running on WSO2 API Manager. API development is started with a Swagger specification with a contract-first approach. Please see the full [Swagger definition](https://raw.githubusercontent.com/wso2/carbon-apimgt/v6.6.163/components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/resources/admin-api.yaml), which is written using Swagger 2.0. This can be also retrieved from the Web app itself using the URL `https://<host-name[:port]>/api/am/admin/v0.16/swagger.json`.
```
Comment (shaniR) [inline en/docs/develop/product-apis/gateway-apis/gateway-v1/guide.md:5]: This can be rewritten as follows : - The REST API is implemented based on the REST specification and best practices as a CXF REST web application running on WSO2 API Manager. A contract-first approach could be utilized for API development starting with a Open API specification. See the admin APIs  [Swagger definition](https://raw.githubusercontent.com/wso2/carbon-apimgt/v6.6.163/components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/resources/admin-api.yaml), which is written using Open API 3.0 .Alternatively, this API definition can be accessed using the product URL as well `https://<host-name[:port]>/api/am/admin/v0.16/swagger.json`.
Note to use the correct OPen API definition URL and the product API version URL for the latest product version

File: en/docs/develop/product-apis/gateway-apis/gateway-v1/guide.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+{
+   "callBackURL":"www.google.lk",
+   "jsonString":"{ \"username\":\"admin\", \"redirect_uris\":\"www.google.lk\", \"client_name\":\"admin_rest_api_admin\", \"grant_types\":\"authorization_code password refresh_token iwa:ntlm urn:ietf:params:oauth:grant-type:saml2-bearer client_credentialsimplicit\" }",
+   "clientName":null,
+   "clientId":"HfEl1jJPdg5tbtrxhAwybN05QGoa",
+   "clientSecret":"l6c0aoLcWR3fwezHhc7XoGOht5Aa"
+}
+```
+
+During the API invocation process request, click the CXF handler first, which calls an introspection API to validate the token. Generate the access token using the already created OAuth application. A sample call to generate the access token is shown below.
```
Comment (shaniR) [inline en/docs/develop/product-apis/gateway-apis/gateway-v1/guide.md:39]: When an API is invoked, the handler invokes an introspection API to validate the token.  

The flow of events are not in order. Can you please revsit it?

File: en/docs/develop/product-apis/gateway-apis/gateway-v1/guide.md [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+{
+   "scope":"apim:tier_view",
+   "token_type":"Bearer",
+   "expires_in":3600,
+   "refresh_token":"33c3be152ebf0030b3fb76f2c1f80bf8",
+   "access_token":"292ff0fd256814536baca0926f483c8d"
+}
+```
+
+Now, you have a valid access token, which you can use to invoke an API. Navigate through the API descriptions to find the required API, obtain an access token as described above and invoke the API with the authentication header. If you use a different authentication mechanism, this process may change.
```
Comment (shaniR) [inline en/docs/develop/product-apis/gateway-apis/gateway-v1/guide.md:59]: what if the process changes?

File: en/mkdocs.yml [hunk 13]
```diff
@@ -295,11 +295,12 @@ nav:
           - Product REST APIs:
                 - RESTful APIs: develop/product-apis/restful-apis.md
                 - Getting Started:
-                    - Publisher API v1: develop/product-apis/getting-started/guide-publisher-v0.16.md 
+                    - Publisher API v1: develop/product-apis/getting-started/guide-publisher-v0.16.md
```
Comment (shaniR) [inline en/mkdocs.yml:298]: we have two publisher API's of the same version. Is ths correct?

File: en/mkdocs.yml [hunk 14]
```diff
@@ -699,4 +703,4 @@ extra:
         - type: linkedin
           link: https://www.linkedin.com/company/wso2
     site_version: 3.2.0
-    base_path: https://apim.docs.wso2.com/en/next
+    base_path: https://apim.docs.wso2.com/en/next
```
Comment (shaniR) [inline en/mkdocs.yml:706]: base_bath should always be a versioned URL. please add the versioned URL.

File: en/docs/develop/product-apis/gateway-apis/gateway-v1/guide.md (hunk 9 above)
Comment (Sarangan0219) [reply en/docs/develop/product-apis/gateway-apis/gateway-v1/guide.md]: https://github.com/wso2/docs-apim/pull/1527/commits/46b28b730c7fd76e644b5b021cac43b057a7b265

File: en/docs/develop/product-apis/gateway-apis/gateway-v1/guide.md (hunk 10 above)
Comment (Sarangan0219) [reply en/docs/develop/product-apis/gateway-apis/gateway-v1/guide.md]: https://github.com/wso2/docs-apim/pull/1527/commits/46b28b730c7fd76e644b5b021cac43b057a7b265

File: en/docs/develop/product-apis/gateway-apis/gateway-v1/guide.md (hunk 12 above)
Comment (Sarangan0219) [reply en/docs/develop/product-apis/gateway-apis/gateway-v1/guide.md]: Instead of Refresh Token grant user can use password grant ,JWT Grant etc according to his/her scope. These details are specified in https://apim.docs.wso2.com/en/next/learn/api-security/oauth2/grant-types/overview/

File: en/mkdocs.yml (hunk 13 above)
Comment (Sarangan0219) [reply en/mkdocs.yml]: https://github.com/wso2/docs-apim/pull/1527/commits/7eb51e9d35eabdd9d931a5fc427d90a1b519c011

**Comment by Mariangela** [comment]
@Sarangan0219 , please check the merge conflicts.

File: en/docs/develop/product-apis/gateway-apis/gateway-v1/guide.md (hunk 12 above)
Comment (shaniR) [reply en/docs/develop/product-apis/gateway-apis/gateway-v1/guide.md]: Shall we reword it to "Refer to [grant types](https://apim.docs.wso2.com/en/next/learn/api-security/oauth2/grant-types/overview/) supported by WSO2 API-M, to learn more on how to use them in place of the Refresh Token Grant. "

File: en/mkdocs.yml (hunk 14 above)
Comment (shaniR) [reply en/mkdocs.yml:712]: please add https://apim.docs.wso2.com/en/3.2.0

File: en/docs/develop/product-apis/getting-started/guide-gateway-v1.md [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+{
+   "scope":"apim:tier_view",
+   "token_type":"Bearer",
+   "expires_in":3600,
+   "refresh_token":"33c3be152ebf0030b3fb76f2c1f80bf8",
+   "access_token":"292ff0fd256814536baca0926f483c8d"
+}
+```
+
+Now, you have a valid access token, which you can use to invoke an API. Navigate through the API descriptions to find the required API, obtain an access token as described above and invoke the API with the authentication header. If you use a different authentication mechanism, this process may change.
```
Comment (shaniR) [inline en/docs/develop/product-apis/getting-started/guide-gateway-v1.md:63]: Now, you have a valid access token which you can use to invoke an API. If you wish to use a different grant type refer to the [Grant types] (https://apim.docs.wso2.com/en/next/learn/api-security/oauth2/grant-types/overview/) document to learn more.

File: en/docs/develop/product-apis/getting-started/guide-gateway-v1.md [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+WSO2 API Manager is a complete API Management solution that addresses all aspects of API management including API development, productization, consumption, security, rate limitation, monetization and analytics.
+
+
+The REST API is implemented based on the REST specification and best practices as a CXF REST web application running on WSO2 API Manager. A contract-first approach could be utilized for API development starting with a Open API specification. 
+See the Gateway API's [Swagger definition](https://raw.githubusercontent.com/wso2/carbon-apimgt/v6.6.163/components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml) , which is written using Open API 3.0 .
+Alternatively, this API definition can be accessed using the product URL as well `https://<host-name[:port]>/api/am/gateway/v1/swagger.json`.
+
+
+The API comes with a pluggable security mechanism. Since API security is implemented as a CXF handler, if you need to plug a custom security mechanism, you can write your own handler and add it to the web service.
+
```
Comment (shaniR) [inline en/docs/develop/product-apis/getting-started/guide-gateway-v1.md:12]: Add this section "In the default flow, when an API is invoked with an access token, first the CXF handler gets engaged and calls an introspection API to validate the token"

File: en/docs/develop/product-apis/restful-apis.md [hunk 17]
```diff
@@ -26,6 +26,11 @@ There are two versions of REST APIs available for the API Publisher and Develope
   <tr>
     <td nowrap="true"><a href="{{base_path}}/develop/product-apis/getting-started/guide-admin-v1/">WSO2 Admin API v0.16</a></td>
     <td> <b>WSO2 recommends that you use the Admin v0.16 REST APIs</b> as it is up to date and has support for the latest WSO2 API Manager 3.2.0 features with regard to the Admin Portal.</td>
+    <td nowrap="true"><a href="{{base_path}}/develop/product-apis/admin-apis/admin-v0.16/admin-v0.16/">WSO2 Admin API v0.16</a></td>
+    <td> <b>WSO2 recommends that you use the Admin v0.16 REST APIs</b> as it is up to date and has support for the latest WSO2 API Manager 3.1.0 features with regard to the Admin Portal.</td>
```
Comment (shaniR) [inline en/docs/develop/product-apis/restful-apis.md:30]: API Manager 3.2.0

File: en/docs/develop/product-apis/getting-started/guide-gateway-v1.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (Sarangan0219) [reply en/docs/develop/product-apis/getting-started/guide-gateway-v1.md]: https://github.com/wso2/docs-apim/pull/1527/commits/88b578c7e4c1e51f1c27511db5e009b534e8677c

File: en/docs/develop/product-apis/getting-started/guide-gateway-v1.md [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+{
+   "callBackURL":"www.google.lk",
+   "jsonString":"{ \"username\":\"admin\", \"redirect_uris\":\"www.google.lk\", \"client_name\":\"admin_rest_api_admin\", \"grant_types\":\"authorization_code password refresh_token iwa:ntlm urn:ietf:params:oauth:grant-type:saml2-bearer client_credentialsimplicit\" }",
+   "clientName":null,
+   "clientId":"HfEl1jJPdg5tbtrxhAwybN05QGoa",
+   "clientSecret":"l6c0aoLcWR3fwezHhc7XoGOht5Aa"
+}
+```
+
+During the API invocation process request, click the CXF handler first, which calls an introspection API to validate the token. Generate the access token using the already created OAuth application. A sample call to generate the access token is shown below.
```
[the remark this answers is not in the document]
Comment (Sarangan0219) [reply en/docs/develop/product-apis/getting-started/guide-gateway-v1.md]: https://github.com/wso2/docs-apim/pull/1527/commits/88b578c7e4c1e51f1c27511db5e009b534e8677c

File: en/docs/develop/product-apis/getting-started/guide-gateway-v1.md [hunk 19]
```diff
@@ -0,0 +1,63 @@
...
+
+Note: Access token must be generated using correct scope for the resource. Scope for each resource is given in resource documentation.
+
+```
```
[the remark this answers is not in the document]
Comment (Sarangan0219) [reply en/docs/develop/product-apis/getting-started/guide-gateway-v1.md:49]: https://github.com/wso2/docs-apim/pull/1527/commits/88b578c7e4c1e51f1c27511db5e009b534e8677c

File: en/docs/develop/product-apis/getting-started/guide-gateway-v1.md [hunk 20]
```diff
@@ -0,0 +1,63 @@
...
+curl -k -d "grant_type=password&username=admin&password=admin&scope=apim:tier_view" -H "Authorization: Basic SGZFbDFqSlBkZzV0YnRyeGhBd3liTjA1UUdvYTpsNmMwYW9MY1dSM2Z3ZXpIaGM3WG9HT2h0NUFh" https://localhost:8243/token
+```
+
+Token response:
```
[the remark this answers is not in the document]
Comment (Sarangan0219) [reply en/docs/develop/product-apis/getting-started/guide-gateway-v1.md:55]: https://github.com/wso2/docs-apim/pull/1527/commits/88b578c7e4c1e51f1c27511db5e009b534e8677c

File: en/docs/develop/product-apis/getting-started/guide-gateway-v1.md [hunk 21]
```diff
@@ -0,0 +1,63 @@
...
+
+
+The REST API is implemented based on the REST specification and best practices as a CXF REST web application running on WSO2 API Manager. A contract-first approach could be utilized for API development starting with a Open API specification. 
+See the Gateway API's [Swagger definition](https://raw.githubusercontent.com/wso2/carbon-apimgt/v6.6.163/components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/resources/gateway-api.yaml) , which is written using Open API 3.0 .
+Alternatively, this API definition can be accessed using the product URL as well `https://<host-name[:port]>/api/am/gateway/v1/swagger.json`.
+
+
+The API comes with a pluggable security mechanism. Since API security is implemented as a CXF handler, if you need to plug a custom security mechanism, you can write your own handler and add it to the web service.
+
+Before invoking the API with the access token, obtain the consumer key/secret key pair by calling the dynamic client registration endpoint. You can request an access token with the preferred grant type. An example is shown below,
```
[the remark this answers is not in the document]
Comment (Sarangan0219) [reply en/docs/develop/product-apis/getting-started/guide-gateway-v1.md]: https://github.com/wso2/docs-apim/pull/1527/commits/88b578c7e4c1e51f1c27511db5e009b534e8677c

File: en/docs/develop/product-apis/getting-started/guide-gateway-v1.md [hunk 22]
```diff
@@ -0,0 +1,63 @@
...
+```
+{
+   "callBackURL":"www.google.lk",
+   "jsonString":"{ \"username\":\"admin\", \"redirect_uris\":\"www.google.lk\", \"client_name\":\"admin_rest_api_admin\", \"grant_types\":\"authorization_code password refresh_token iwa:ntlm urn:ietf:params:oauth:grant-type:saml2-bearer client_credentialsimplicit\" }",
+   "clientName":null,
+   "clientId":"HfEl1jJPdg5tbtrxhAwybN05QGoa",
+   "clientSecret":"l6c0aoLcWR3fwezHhc7XoGOht5Aa"
+}
+```
+
```
[the remark this answers is not in the document]
Comment (Sarangan0219) [reply en/docs/develop/product-apis/getting-started/guide-gateway-v1.md:39]: https://github.com/wso2/docs-apim/pull/1527/commits/88b578c7e4c1e51f1c27511db5e009b534e8677c

File: en/docs/develop/product-apis/getting-started/guide-gateway-v1.md [hunk 23]
```diff
@@ -0,0 +1,63 @@
+# Getting Started
+
+WSO2 API Manager is a complete API Management solution that addresses all aspects of API management including API development, productization, consumption, security, rate limitation, monetization and analytics.
+
```
[the remark this answers is not in the document]
Comment (Sarangan0219) [reply en/docs/develop/product-apis/getting-started/guide-gateway-v1.md:4]: https://github.com/wso2/docs-apim/pull/1527/commits/dfc6cce75030fb2c093887ef3c5342fc7a9bce21

File: en/docs/develop/product-apis/getting-started/guide-gateway-v1.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (Sarangan0219) [reply en/docs/develop/product-apis/getting-started/guide-gateway-v1.md]: https://github.com/wso2/docs-apim/pull/1527/commits/dfc6cce75030fb2c093887ef3c5342fc7a9bce21

File: en/docs/develop/product-apis/getting-started/guide-gateway-v1.md (hunk 15 above)
Comment (Sarangan0219) [reply en/docs/develop/product-apis/getting-started/guide-gateway-v1.md]: https://github.com/wso2/docs-apim/pull/1527/commits/dfc6cce75030fb2c093887ef3c5342fc7a9bce21

File: en/docs/develop/product-apis/getting-started/guide-gateway-v1.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (Sarangan0219) [reply en/docs/develop/product-apis/getting-started/guide-gateway-v1.md:12]: https://github.com/wso2/docs-apim/pull/1527/commits/a8d04065338ebc31af3e2f8f63d9c389a2c93370

Comment (Sarangan0219) [reply en/docs/develop/product-apis/getting-started/guide-gateway-v1.md:12]: https://github.com/wso2/docs-apim/pull/1527/commits/dfc6cce75030fb2c093887ef3c5342fc7a9bce21

File: en/docs/develop/product-apis/restful-apis.md (hunk 17 above)
Comment (Sarangan0219) [reply en/docs/develop/product-apis/restful-apis.md]: Already Fixed

File: en/docs/develop/product-apis/gateway-apis/gateway-v1/guide.md (hunk 12 above)
Comment (Sarangan0219) [reply en/docs/develop/product-apis/gateway-apis/gateway-v1/guide.md]: Fixed

**Review by shaniR — APPROVED** [review]
